rocAL/source/augmentations/filter_augmentations/node_erase.cpp

```diff
@@ -0,0 +1,336 @@
+    }
+
+    // Anchor tensor handle
+    vx_size anchor_dims[2]   = { (vx_size)_total_boxes, (vx_size)4 };
+    vx_size anchor_stride[2] = { 0, 0 };
+    anchor_stride[0] = sizeof(vx_int32);
+    anchor_stride[1] = anchor_stride[0] * anchor_dims[0];
+
+    size_t bytes_n = anchor_stride[1] * anchor_dims[1];
+    allocate_host_or_pinned_mem(&_anchor_ptr, bytes_n, mem_type);
+    std::memcpy(_anchor_ptr, _anchor_vec.data(), bytes_n);
+
+    vx_tensor _anchor_vx = vxCreateTensorFromHandle(vxGetContext((vx_reference)_graph->get()),
+                                            2, anchor_dims, VX_TYPE_INT32, 0, anchor_stride, _anchor_ptr, vx_mem);
+    if (!_anchor_vx) THROW("vxCreateTensorFromHandle for anchor tensor failed");
+    {
+        vx_status s = vxGetStatus((vx_reference)_anchor_vx);
+        if (s != VX_SUCCESS) THROW("vxCreateTensorFromHandle anchor failed: " + TOSTR(s));
+    }
+
+    // Color tensor handle - match input tensor data type
+    auto input_data_type = _inputs[0]->info().data_type();
+    vx_enum vx_color_type = interpret_tensor_data_type(input_data_type);
+    size_t color_element_size = tensor_data_size(input_data_type);
+    
+    vx_size color_dims[2]   = { (vx_size)_total_boxes, (vx_size)_inputs[0]->info().get_channels() };
+    vx_size color_stride[2] = { 0, 0 };
+    color_stride[0] = color_element_size;
+    color_stride[1] = color_stride[0] * color_dims[0];
+
+    size_t bytes_c = color_stride[1] * color_dims[1];
+    allocate_host_or_pinned_mem(&_color_ptr, bytes_c, mem_type);
+    
+    // Convert and copy fill values according to input data type
+    if (input_data_type == RocalTensorDataType::UINT8) {
+        uint8_t* color_ptr_u8 = static_cast<uint8_t*>(_color_ptr);
+        for (size_t i = 0; i < _fill_values_vec.size(); i++) {
+            color_ptr_u8[i] = static_cast<uint8_t>(std::round(std::max(0.0f, std::min(255.0f, _fill_values_vec[i]))));
+        }
+    } else if (input_data_type == RocalTensorDataType::INT16) {
+        int16_t* color_ptr_i16 = static_cast<int16_t*>(_color_ptr);
+        for (size_t i = 0; i < _fill_values_vec.size(); i++) {
+            color_ptr_i16[i] = static_cast<int16_t>(std::round(_fill_values_vec[i]));
+        }
+    } else if (input_data_type == RocalTensorDataType::UINT32) {
+        uint32_t* color_ptr_u32 = static_cast<uint32_t*>(_color_ptr);
+        for (size_t i = 0; i < _fill_values_vec.size(); i++) {
+            color_ptr_u32[i] = static_cast<uint32_t>(std::round(std::max(0.0f, _fill_values_vec[i])));
+        }
+    } else if (input_data_type == RocalTensorDataType::INT32) {
+        int32_t* color_ptr_i32 = static_cast<int32_t*>(_color_ptr);
+        for (size_t i = 0; i < _fill_values_vec.size(); i++) {
+            color_ptr_i32[i] = static_cast<int32_t>(std::round(_fill_values_vec[i]));
+        }
+    } else if (input_data_type == RocalTensorDataType::FP16) {
+        // For FP16, we need to handle it as 16-bit values
```

Comment:
(blocking) For FP16 we cannot `static_cast<uint16_t>` the float – that just truncates to an integer and produces completely wrong IEEE-754 half bit patterns (e.g. 1.0 becomes 0x0001 instead of 0x3C00). Please run the values through a real float→half conversion helper (there are existing utilities in rocAL, e.g. hip half conversion or `float_to_half`), otherwise the erase colors are garbage for FP16 tensors.